napari/settings/_base.py

```diff
@@ -364,8 +367,32 @@ def config_file_settings_source(
 
     # if the config has a `sources` list, read those too and merge.
     sources = list(getattr(settings.__config__, 'sources', []))
-    if config_path:
+    if config_path and Path(config_path).exists():
         sources.append(config_path)
+    elif config_path and Path(config_path).parent.parent.exists():
+        # check for previous version directory
+        napari_dir = Path(config_path).parent.parent
+        napari_versions = (
+            (version.Version(dir.name), dir)
+            for dir in napari_dir.iterdir()
+            if dir.is_dir()
+            and isinstance(version.parse(dir.name), version.Version)
```

Comment:
I think `version.parse` can raise here, but we never expect it to? Because all sub-directories should be parsable versions?

Reply:
We do not control user drives and plugins so it may happen that some random "aaa" directory will be present here. 

For example if someone is using PartSega as a napari plugin there will be present `PartSeg_napari_plugins` directory next to `settings.yaml` file from current stable version.

Reply:
I've tested extensively with various dir names. I've not been able to get it to raise the `InvalidVersion`. For example:
`packaging.version.parse("test_dir")` 
returns: <LegacyVersion('test_dir')>
Few others:
```
In [4]: version.parse("test_dir1111")
Out[4]: <LegacyVersion('test_dir1111')>

In [5]: version.parse("aaa1111")
Out[5]: <LegacyVersion('aaa1111')>

In [6]: version.parse("__aaa1111")
Out[6]: <LegacyVersion('__aaa1111')>
```

Reply:
Yeah, I could make it break with any strings. Maybe it only raises for some unsupported non-string inputs.

Reply:
If I'm reading the docs right, any `str` should get the `LegacyVersion` version:
https://packaging.pypa.io/en/stable/version.html#packaging.version.LegacyVersion

Reply:
Sorry, PBCAK.